### PR TITLE
Add support for Ruby 3. Fixes #36

### LINKED ISSUE #36: xumlidot doesn't work right with Ruby 3
... because the `sexp_processor` is broken in ruby 3.

I've made a [PR](https://github.com/seattlerb/sexp_processor/issues/31#issue-955054865).

### PR DESCRIPTION
**Pull request by leoarnold**

### CODE REVIEW

**Review by os6sense — DISMISSED** [review]

**Comment by leoarnold** [comment]
I'll have a look at the failing build.

**Comment by os6sense** [comment]
> I'll have a look at the failing build.

Many thanks; apologies to have left things waiting for merge for so long, I've been trying to free up some time to look at it.

I've merged a few pending PR's which may be the root cause.

**Comment by os6sense** [comment]
https://github.com/os6sense/xumlidot/commit/568678e7813a1f0c2fee5b9cc66bb6faaaf73f3e seems that merge
 is causing issues

**Comment by leoarnold** [comment]
No worries, I'll find it.

**Comment by leoarnold** [comment]
The fix is up (see #45). There may be some issue in the setup of the service you are using for the RuboCop build job.

**Comment by os6sense** [comment]
Pronto has given me more hassle than it is worth I must admit. I'll ignore the rubocop issues and address separately; many thanks @leoarnold

File: spec/integration/lib_methods/method_name_escaping_spec.rb [hunk 1]
```diff
@@ -3,7 +3,7 @@
 require 'spec_helper'
 
 describe 'Output of lib/methods' do
-  let(:directory) { 'spec/app/lib/methods/' }
+  let(:directory) { 'spec/app/lib/methods/method_name_escaping.rb' }
```
Comment (os6sense) [inline spec/integration/lib_methods/method_name_escaping_spec.rb:6]: :facepalm: :+1:

**Review by os6sense — APPROVED** [review]

**Pull request merged**